### PR TITLE
Add ``deprecate_settings`` to bears

### PR DESCRIPTION
**Pull request by Asnelchristian**

### CODE REVIEW

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/8ba353e16f235f905e7b19909041cee1144b5221

File: bears/js/JSHintBear.py [hunk 1]
```diff
@@ -280,7 +316,7 @@ def generate_config(filename, file,
             code must adhere to.
         """
         if not jshint_config:
-            options = {"bitwise": allow_bitwise_operators,
+            options = {"bitwise": not allow_bitwise_operators,
```
Comment (sils1297) [inline bears/js/JSHintBear.py]: hey, why are you changing functionality here?

Comment (sils1297) [inline bears/js/JSHintBear.py]: if it's needed, put it in an own commit

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/2ea4e619b694fc3c97fc9ed70546a521469a8330

File: bears/js/JSONFormatBear.py [hunk 2]
```diff
@@ -19,6 +20,7 @@ class JSONFormatBear(LocalBear):
     LICENSE = 'AGPL-3.0'
     CAN_DETECT = {'Formatting'}
 
+    @deprecate_settings(escape_unicode='keep_unicode')
```
Comment (sils1297) [inline bears/js/JSONFormatBear.py]: that needs a lambda doesn't it?

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/2ea4e619b694fc3c97fc9ed70546a521469a8330

File: bears/natural_language/WriteGoodLintBear.py [hunk 3]
```diff
@@ -22,6 +23,14 @@ class WriteGoodLintBear:
...
 
     @staticmethod
+    @deprecate_settings(allow_passive_voice='check_passive_voice',
+                        allow_so_beginning='check_so_beginning',
+                        allow_adverbs='check_adverbs',
+                        allow_repeated_words='check_repeated_words',
+                        allow_there_is='check_there_is',
+                        allow_ambiguous_words='check_ambiguos_words',
+                        allow_extra_words='check_extra_words',
+                        allow_cliche_phrases='check_cliche_exists')
```
Comment (sils1297) [inline bears/natural_language/WriteGoodLintBear.py]: don't all of those need inverts? Btw. I think you'll want to make an invert function somewhere and import it instead of doing a lambda every time?

File: bears/js/JSHintBear.py (hunk 1 above)
Comment (Asnelchristian) [inline bears/js/JSHintBear.py]: okay :)

File: bears/natural_language/WriteGoodLintBear.py (hunk 3 above)
Comment (Asnelchristian) [inline bears/natural_language/WriteGoodLintBear.py]: no they don't, yeah right (should i do that i coala-utils?)

Comment (sils1297) [inline bears/natural_language/WriteGoodLintBear.py]: yeah coala-utils makes sense, like param conversion utils or so

File: bears/js/JSONFormatBear.py (hunk 2 above)
Comment (Asnelchristian) [inline bears/js/JSONFormatBear.py]: lemme check

Comment (Asnelchristian) [inline bears/js/JSONFormatBear.py]: it doesn't

Comment (sils1297) [inline bears/js/JSONFormatBear.py]: why? escaping unicode is **not** keeping it

File: bears/natural_language/WriteGoodLintBear.py (hunk 3 above)
Comment (sils1297) [inline bears/natural_language/WriteGoodLintBear.py]: why not? Allowing ... is *not* checking that ... isn't there so it's inverted, isn't it?

Comment (Asnelchristian) [inline bears/natural_language/WriteGoodLintBear.py]: that is so TRUE
making some changes for that t happen :)

File: bears/js/JSHintBear.py [hunk 4]
```diff
@@ -40,6 +42,33 @@ class JSHintBear:
...
+                        max_parameters='maxparams',
+                        allow_missing_semicolon='allow_missing_semicol',
+                        allow_this_statements='allow_this_stmt',
+                        allow_with_statements='allow_with_stmt',
+                        allow_bitwise_operators=('prohibit_bitwise', negate),
+                        max_statements='maxstatements',
+                        max_depth='maxdepth',
+                        allow_comma_operator=('prohibit_comma', negate),
+                        allow_non_breaking_whitespace=(
+                            'prohibit__non_breaking_whitespace', negate),
```
Comment (sils1297) [inline bears/js/JSHintBear.py]: really? With a double underscore?

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/4c19b1a83f1a605b312e22bbb04c2b2207b563cd

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/05d3f3f9abf9206fff3cdc9afbb74471e3626b00

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala-bears/pull/676/commits/71378ad75f5ab96518dcf58ce26678799a387bd4#r74884931, see comment

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/753de6c0dc36c4ea4111ea0f06f21424627657d6

File: bears/js/JSONFormatBear.py (hunk 2 above)
Comment (Asnelchristian) [inline bears/js/JSONFormatBear.py]: once more you are right :) :+1:

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/676/commits/1da36879d7956185e01e33d9b868c697b54da8e0 https://github.com/coala-analyzer/coala-bears/pull/676/commits/6fc858a2284933140b3255dd0bb7939cd5962b52 https://github.com/coala-analyzer/coala-bears/pull/676/commits/b96fb9d0825e26bc5b8123664387c8d8d7f95cfe

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/5385dd2db8114054bdfea248a71d7726427f1dfd

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/abb2e822698fa750b69dd39b4361c0a9e6d040ed

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/97a4455f288b13634ea843d579cd578bc28fb008

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/73d1b8f3b6718d3dfeb53fe42f68287f74161104

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/7a29df2b9ceb4c4a7961b6169a28db73c6b0557c https://github.com/coala-analyzer/coala-bears/pull/676/commits/8b1fbeda4e7baf295a8567b52224dedde08d864c

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/b901af9988750eeb34d202331c8ebd1e01702cdc

File: bears/java/JavaPMDBear.py [hunk 5]
```diff
@@ -32,6 +33,8 @@ def check_prerequisites(cls):  # pragma: no cover
             return True
 
     @staticmethod
+    @deprecate_settings(allow_unnecessary_code='check_unnecessary',
+                        allow_unused_code='check_unused')
```
Comment (sils1297) [inline bears/java/JavaPMDBear.py]: don't those need negation? :/

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/ab9a2902eb568f4fc52e282106593a2f0fd50683

**Comment by sils1297** [comment]
unack https://github.com/coala-analyzer/coala-bears/pull/676/commits/b57963571d53f1c24b13990c7b351af3ce36825c

File: bears/js/JSHintBear.py [hunk 6]
```diff
@@ -40,6 +42,33 @@ class JSHintBear:
...
+                        allow_typeof=('prohibit_typeof', negate),
+                        allow_var_statement=(
+                            'prohibit_variable_statements', negate),
+                        allow_grouping_operator=('prohibit_groups', negate),
+                        allow_variable_shadowing='shadow',
+                        use_mozilla_extension='using_mozilla',
+                        allow_constructor_functions=('prohibit_new', negate),
+                        allow_argument_caller_and_callee=(
+                            'prohibit_arg', negate),
+                        allow_iterator_property='iterator',
```
Comment (sils1297) [inline bears/js/JSHintBear.py]: do those need negation? Did iterator enable the check?

Comment (Asnelchristian) [inline bears/js/JSHintBear.py]: it needs it, that is just a mistake

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/676/commits/7ce7aee55867c3d513a04b1f5f73e8e6d127d9cc  https://github.com/coala-analyzer/coala-bears/pull/676/commits/ec8aeb19c225293735159b77d7f7eae3cf9daeb6 https://github.com/coala-analyzer/coala-bears/pull/676/commits/35fb7984d348216f1406f6e1134fe9bcd6c83895 https://github.com/coala-analyzer/coala-bears/pull/676/commits/62fb9d71d931e88d92f7f60305bae5a7fc11a0c2  https://github.com/coala-analyzer/coala-bears/pull/676/commits/270f1621591b2100766ed2cb5da11d3642c75bcb

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/676/commits/60675a9850e7adb6d9bf9e893ef71042e5639bb2 https://github.com/coala-analyzer/coala-bears/pull/676/commits/92049ff61158d2527b26f13d7841684aa8ca99e6  https://github.com/coala-analyzer/coala-bears/pull/676/commits/14a1bf4a94c1d8f219290ee45a36573c4bdca0f5

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/676/commits/0e42009d6afd9392d357283a496f528d27e8b667 https://github.com/coala-analyzer/coala-bears/pull/676/commits/e06167c61772cc572f36b353ed250c525404fae1 https://github.com/coala-analyzer/coala-bears/pull/676/commits/33a83e42ee3efd6423126ea6b5ba7345f439567b

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/c5d36cf997e6ea1ddb442b6579a4da1ea27de525#diff-732cf5cf5f77985606c5e89def5320bfR71

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/e21ad1abffb0f23154fb87ec14a2f8518abf12b2

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/a85ae5e898a9024b954c45f08f548aff160d1df3

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/676/commits/837acd7e6e4da0b1ea58063789cf89f41d0eded3

**Comment by sils1297** [comment]
@rultor merge
